src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -447,23 +444,57 @@ IpcIoFile::canWait() const
     return false; // do not want to wait that long
 }
 
-/// called when coordinator responds to worker open request
 void
-IpcIoFile::HandleOpenResponse(const Ipc::StrandMessage &response)
+IpcIoFile::HandleStrandReadyResponse(const Ipc::StrandReady &response)
 {
-    debugs(47, 7, HERE << "coordinator response to open request");
-    for (IpcIoFileList::iterator i = WaitingForOpen.begin();
-            i != WaitingForOpen.end(); ++i) {
-        if (response.strand.tag == (*i)->dbName) {
-            (*i)->openCompleted(&response);
-            WaitingForOpen.erase(i);
-            return;
-        }
+    debugs(47, 7, "disker" << response.strand.kidId <<  " " << response.strand.tag << " is ready");
+    if (const auto file = StopWaiting(response.strand))
+        file->openCompleted(&response);
+    else
+        debugs(47, 4, "premature or late disker response");
+}
+
+/// reschedules IpcIoFile open request timeout for the head of the waiting queue
```

Comment:
This function does not (necessarily) REschedule a timeout -- it adds a new timeout. It also does not place the new timeout onto the _head_ of the queue -- the timeout will actually go to the _tail_ of the queue. Finally, the head/tail detail is unimportant for the caller -- they just want to schedule a timeout.

```suggestion
/// (re)start monitoring IpcIoFile open request timeout
```

Reply:
Ah, got it, thank you. I think we can omit that implementation detail from this method description. Resolving.